server/startup/migrations/v243.ts

```diff
@@ -0,0 +1,43 @@
+import { addMigration } from '../../lib/migrations';
+import { Uploads } from '../../../app/models/server';
+
+
+addMigration({
+	version: 243,
+	up() {
+		const images = Uploads.model.rawCollection().find({ typeGroup: 'image' });
+		const getAspectRatio = (image: Record<string, any>): number => image.identify.size.width / image.identify.size.height;
+
+		// proportion rate error margin 7%
+		const epsilon = 0.07;
+
+		// 2 seconds
+		const thumbCreationTimeout = 2000;
+
+		images.forEach(
```

Comment:
since you're using `rawCollection` this `forEach` is `async`, and because of that the migration is "finished" **before** it loops over all documents, so the migration keeps running in background even after the servers starts, which is not ideal. I recommend using `fibers/future` to lock the migration during the execution.